File: packages/client_react/src/types/controlCode.ts
export interface ControlCodeDto {
  id: number;
  materialId: number;
  code: string;
  isReceptionReceived: boolean;
}

export interface ControlCodeSelectionDto {
  id: number;
  code: string;
}

export interface CreateControlCodeDto {
  materialId: number;
  code: string;
}

File: server/QCLab/Models/ControlCode.cs
using System;
using System.Collections.Generic;

namespace QCLab.Models;

public partial class ControlCode
{
    public long Id { get; set; }

    public long MaterialId { get; set; }

    public string Code { get; set; } = null!;

    public bool IsReceptionReceived { get; set; }

    public DateTime DateCreated { get; set; }

    public virtual ICollection<Certificate> Certificates { get; set; } = new List<Certificate>();

    public virtual Material Material { get; set; } = null!;

    public virtual ReagentLot? ReagentLot { get; set; }

    public virtual ICollection<Reception> Receptions { get; set; } = new List<Reception>();
}
